src/write_prometheus_test.c

```diff
@@ -279,11 +279,71 @@ DEF_TEST(format_metric_family) {
                       },
               },
           .want = "# HELP unit_test\n"
-                  "# TYPE unit_test untyped\n"
+                  "# TYPE unit_test gauge\n"
                   "unit_test{job=\"service name\",instance=\"service instance "
                   "id\",metric_name=\"unit.test\"} 42\n"
                   "\n",
       },
+      {
+          .name = "METRIC_TYPE_COUNTER_FP",
+          .fam =
+              {
+                  .name = "unit_test",
+                  .type = METRIC_TYPE_COUNTER_FP,
+                  .metric =
+                      {
+                          .ptr =
+                              &(metric_t){
+                                  .value.counter_fp = 42.0,
+                              },
+                          .num = 1,
+                      },
+              },
+          .want = "# HELP unit_test_total\n"
+                  "# TYPE unit_test_total counter\n"
+                  "unit_test_total 42\n"
+                  "\n",
+      },
+      {
+          .name = "METRIC_TYPE_UP_DOWN",
+          .fam =
+              {
+                  .name = "unit_test",
+                  .type = METRIC_TYPE_UP_DOWN,
+                  .metric =
+                      {
+                          .ptr =
+                              &(metric_t){
+                                  .value.up_down = 42,
+                              },
+                          .num = 1,
+                      },
+              },
+          .want = "# HELP unit_test\n"
+                  "# TYPE unit_test gauge\n"
+                  "unit_test 42\n"
+                  "\n",
+      },
+      {
+          .name = "METRIC_TYPE_UP_DOWN_FP",
+          .fam =
+              {
+                  .name = "unit_test",
+                  .type = METRIC_TYPE_UP_DOWN_FP,
+                  .metric =
+                      {
+                          .ptr =
+                              &(metric_t){
+                                  .value.up_down_fp = 42.0,
+                              },
+                          .num = 1,
+                      },
+              },
+          .want = "# HELP unit_test\n"
+                  "# TYPE unit_test gauge\n"
+                  "unit_test 42\n"
+                  "\n",
+      },
```

Comment:
These checks could include (negative) test for UNTYPED metric.

Reply:
+1, but unrelated to this PR.